cypress/integration/pages/onDemandRadio/tests.js

```diff
@@ -53,5 +55,97 @@ export default ({ service, pageType, variant, isAmp }) => {
         });
       });
     });
+    describe(`Tests for ${service} ${pageType} ${variant} with toggle use`, () => {
+      before(() => {
+        cy.getToggles(config[service].name);
+      });
+      describe('Recent Episodes component', () => {
+        it('should be displayed if the toggle is on, and shows the expected number of items (does not run on local)', function test() {
+          // I can enable tests running on local when I have updated all the fixture data to show the component
+          if (Cypress.env('APP_ENV') === 'local') {
+            cy.log('Does not run on local');
+          } else {
+            cy.fixture(`toggles/${config[service].name}.json`).then(toggles => {
+              const recentEpisodesEnabled = path(
+                ['recentAudioEpisodes', 'enabled'],
+                toggles,
+              );
+              cy.log(
+                `Recent Episodes component enabled? ${recentEpisodesEnabled}`,
+              );
+              // There cannot be more episodes shown than the max allowed
+              if (recentEpisodesEnabled) {
+                const recentEpisodesMaxNumber = path(
+                  ['recentAudioEpisodes', 'value'],
+                  toggles,
+                );
+                const currentPath = Cypress.env('currentPath');
+                const url =
+                  Cypress.env('APP_ENV') === 'test'
+                    ? `${currentPath}?renderer_env=live`
+                    : `${currentPath}`;
+
+                cy.request(getDataUrl(url)).then(({ body }) => {
+                  const numberOfEpisodesinData =
+                    body.relatedContent.groups[0].promos.length;
+                  // There cannot be more episodes than the number present in the data
+                  const numberOfEpisodesInDataUnderLimit = Math.min(
+                    numberOfEpisodesinData,
+                    recentEpisodesMaxNumber,
+                  );
+
+                  // Count the number of episodes that are available and so will show (there can be unavailable episodes in the list)
+                  let countAvailableEpisodes = 0;
+                  for (
+                    let i = 0;
+                    i < numberOfEpisodesInDataUnderLimit;
+                    i += 1
+                  ) {
+                    if (
+                      body.relatedContent.groups[0].promos[i].media.versions
+                        .length > 0
+                    ) {
+                      countAvailableEpisodes += 1;
+                    }
+                  }
```

Comment:
```suggestion
                  let countAvailableEpisodes  = body.relatedContent.groups[0].promos.filter(({ media }) => versions.length)
```

Reply:
I think my suggestion above is actually your `expectedNumberOfEpisodes`

Reply:
Nice! I need to remind myself about how filter works. Haven't seen it for a while. Will commit.

Reply:
Aha, so I can also remove the part after this that uses the maxNumberOfEpisodes in the Math.min to get the expectedNumberOfEpisodes . I'll try it

Reply:
Does this also mean I don't need to do 
const numberOfEpisodesinData =
                    body.relatedContent.groups[0].promos.length;
Because this method of counting episodes that have versions, would always give a number equal to or lower than the number of items in the data.

Reply:
yeh I think so - if you have linting set up you can your `expectedNumberOfEpisodes` variable to my suggestions and then delete all the red squiggly code 🙂

Reply:
I'm just doing it in vscode so I can remove other stuff that has become unecessary and watch it run. So far so good

Reply:
as a side note, with javascript array iteration methods, you should almost never have to write a for loop. I haven't written a for loop in years

https://www.w3schools.com/js/js_array_iteration.asp

Reply:
changed :)